# Application_Voyage - Kayak

*jedha Nov. 2023*  
Mélanie Beraud _ melanie.beraud@gmail.com

### API or website to scape:
GPS coordinates: https://nominatim.org/  
Weather: https://openweathermap.org/appid   
Hotels: booking.com

## Imports
### Import Librairies

In [1]:
# import
## import for API use
import requests
import pandas as pd

### Import fichiers et données importantes

In [2]:
top_35 = ["Mont Saint Michel", "St Malo", "Bayeux", "Le Havre", "Rouen", 
          "Paris", "Amiens", "Lille", "Strasbourg", "Chateau du Haut Koenigsbourg",
           "Colmar", "Eguisheim", "Besancon", "Dijon", "Annecy", 
           "Grenoble", "Lyon", "Gorges du Verdon", "Bormes les Mimosas", "Cassis", 
           "Marseille", "Aix en Provence", "Avignon", "Uzes", "Nimes", 
           "Aigues Mortes", "Saintes Maries de la mer", "Collioure", "Carcassonne", "Ariege", 
           "Toulouse", "Montauban", "Biarritz", "Bayonne", "La Rochelle"]


## GPS coordinates

https://nominatim.openstreetmap.org/
    /search - search OSM objects by name or type
    /reverse - search OSM object by their location
    /lookup - look up address details for OSM objects by their ID
    /status - query the status of the server
    /deletable - list objects that have been deleted in OSM but are held back in Nominatim in case the deletion was accidental
    /polygons - list of broken polygons detected by Nominatim
    /details - show internal details for an object (for debugging only)

In [26]:
payload = {'city':'Paris', 'format':'json'}
#payload='Paris' #idem

reponse = requests.get('https://nominatim.openstreetmap.org/search', params=payload)

print('ReponseStatus:\n', reponse) #200
print('Retour en texte:\n', reponse.text) # no gps coordinates
#print('Retour cookies:\n', reponse.cookies)
#print('Retour headers:\n', reponse.headers)
data_r = reponse.json()['adresstype'=='city']

ReponseStatus:
 <Response [200]>
Retour en texte:
 [{"place_id":83293737,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":71525,"lat":"48.8534951","lon":"2.3483915","class":"boundary","type":"administrative","place_rank":12,"importance":0.8317101715588673,"addresstype":"city","name":"Paris","display_name":"Paris, Île-de-France, France métropolitaine, France","boundingbox":["48.8155755","48.9021560","2.2241220","2.4697602"]},{"place_id":83293355,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":7444,"lat":"48.8588897","lon":"2.3200410217200766","class":"boundary","type":"administrative","place_rank":15,"importance":0.8317101715588673,"addresstype":"suburb","name":"Paris","display_name":"Paris, Île-de-France, France métropolitaine, France","boundingbox":["48.8155755","48.9021560","2.2241220","2.4697602"]}]


In [37]:
data_r = reponse.json()['adresstype'=='city']

reponse.text
[{"place_id":83293737,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":71525,"lat":"48.8534951","lon":"2.3483915","class":"boundary","type":"administrative","place_rank":12,"importance":0.8317101715588673,"addresstype":"city","name":"Paris","display_name":"Paris, Île-de-France, France métropolitaine, France","boundingbox":["48.8155755","48.9021560","2.2241220","2.4697602"]},{"place_id":83293355,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":7444,"lat":"48.8588897","lon":"2.3200410217200766","class":"boundary","type":"administrative","place_rank":15,"importance":0.8317101715588673,"addresstype":"suburb","name":"Paris","display_name":"Paris, Île-de-France, France métropolitaine, France","boundingbox":["48.8155755","48.9021560","2.2241220","2.4697602"]}]

In [40]:
data_r

{'place_id': 83293737,
 'licence': 'Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright',
 'osm_type': 'relation',
 'osm_id': 71525,
 'lat': '48.8534951',
 'lon': '2.3483915',
 'class': 'boundary',
 'type': 'administrative',
 'place_rank': 12,
 'importance': 0.8317101715588673,
 'addresstype': 'city',
 'name': 'Paris',
 'display_name': 'Paris, Île-de-France, France métropolitaine, France',
 'boundingbox': ['48.8155755', '48.9021560', '2.2241220', '2.4697602']}

In [43]:
#data_r['adresstype'=='city']['lat']
data_r
coord = {'ville':data_r['name'], 'place_id': data_r['place_id'], 'lat':data_r['lat'],'lon':data_r['lon'], 'display_name':data_r['display_name']}
coord

{'ville': 'Paris',
 'place_id': 83293737,
 'lat': '48.8534951',
 'lon': '2.3483915',
 'display_name': 'Paris, Île-de-France, France métropolitaine, France'}